### DISCUSSION TITLE
Expected YAML input for multiple samples

### DISCUSSION THREAD

**Discussion by maotian06** [uclahs-cds/pipeline-call-sSNV#66] [Ideas]
Two things want to achieve here:
1. include the multiple samples in the call-sSNV - mutect2 algorithms
2. update the input to YAML.

Here is the expected YAML file:
```
---
# Sample inputs
sample_name: ''

# algorithm and reference used
algorithm : [] // 'somaticsniper', 'strelka2', 'mutect2'
reference : ''

# options for specical tumor mode
tumor_only_mode: false
multi_sample_mode: true

# path to BAMs
normal:
  -[id1, path/to/1.bam]
  -[id2, path/to/2.bam]
tumor:
  -[id1, path/to/1.bam]
  -[id2, path/to/2.bam]
```

@tyamaguchi-ucla @yashpatel6 @zhuchcn

**Comment by yashpatel6** [comment]
I would suggest keeping the `algorithm` and `reference` params in the config file and keep only the sample-specific params in the YAML. For `tumor_only_mode` and `multi_sample_mode`, if these will be consistent across most samples, you can have a default value in the config file and add a comment in the YAML that if specified in a sample YAML, they'll over-write the config file param for the specific sample. So maybe something like:
```
---
# Sample inputs
sample_name: ''

# options for specical tumor mode
# If included in sample YAML, the defaults in the config file will be overwritten
tumor_only_mode: false
multi_sample_mode: true

# path to BAMs
normal:
  -[id1, path/to/1.bam]
  -[id2, path/to/2.bam]
tumor:
  -[id1, path/to/1.bam]
  -[id2, path/to/2.bam]
```

**Reply by maotian06** [reply]
Okay! I will add the `algorithm` and `reference` in the YAML! But I also want to keep the `algorithm` and  `reference` in the `template.config`, so the users don't need to define them when they are the same across the cohort.
In this case, their YAML files will be really simple. Something like:
```
---
sample_name: ''
normal:
  -[id1, path/to/1.bam]
  -[id2, path/to/2.bam]
tumor:
  -[id1, path/to/1.bam]
  -[id2, path/to/2.bam]
 ```

**Reply by tnyamaguchi** [reply]
Hi guys,

Some general thoughts. Do we want to use `list` in pipeline YAML inputs? If so, we might have to deal with magic numbers so I think `dictionary` (`key:value`) would be easier to use and more flexible (even with nesting)

We used to use these YAML formats - https://confluence.mednet.ucla.edu/pages/viewpage.action?pageId=77029703 (legacy call-gSNP) and https://confluence.mednet.ucla.edu/display/BOUTROS/Alignment+Pipeline+How-To (legacy align-DNA)

**Reply by yashpatel6** [reply]
Right, I think we do want to use `dictionaries`. Mao and I talked recently and decided to switch it to a `dictionary` format, similar to the example in `call-FusionTranscript`.

**Reply by maotian06** [reply]
@tyamaguchi-ucla I cannot open the link you provide... 
YES! Yash showed me how to use `dictionary` in YAML files and nextflow, and I am trying it out now. Will update the result here!

**Reply by tnyamaguchi** [reply]
@maotian06 you can ask Julie R for access.
